skt/__init__.py

```diff
@@ -368,16 +269,17 @@ def merge_git_ref(self, uri, ref="master"):
         return (0, head)
 
     def merge_patchwork_patch(self, uri):
-        (rpc, patchid) = parse_patchwork_url(uri)
-
-        patchinfo = rpc.patch_get(patchid)
+        # Use os.path for manipulation with URL because urlparse can't deal
+        # with URLs ending both with and without slash.
+        mbox_url = os.path.join(uri, 'mbox')
 
-        if not patchinfo:
-            raise Exception(
-                "Failed to fetch patch info for patch %s" % patchid
-            )
+        try:
+            response = requests.get(mbox_url)
+        except requests.exceptions.RequestException:
+            raise()
 
-        pdata = stringify(rpc.patch_get_mbox(patchid))
+        if response.status_code != requests.codes.ok:
+            raise Exception('Nonexistent patch passed: %s' % uri)
```

Comment:
This error message would be wrong for anything except the 404 response. @veruu, could you perhaps submit a fix changing this to a more general error message? E.g. "Failed to retrieve patch %s: response code %u", or something like that?

Reply:
I'm not sure what other error you can get from patchwork, but I agree that a more generic message would cover it better. Will submit a follow-up patch.

Reply:
Well, there's always the Murphy's law, but things like "500 Internal Server Error", or "504 Gateway Timeout" could happen and then skt would lead user's troubleshooting in the wrong direction.